pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -241,7 +241,11 @@ public static String replaceSpecialCharacterInPropertyValue(String value) {
    * {@link #replaceSpecialCharacterInPropertyValue(String)}.
    */
   public static String recoverSpecialCharacterInPropertyValue(String value) {
-    value = StringEscapeUtils.unescapeJava(value);
+    try {
```

Comment:
I saw a similar change in another method (introduced by the same PR).  I did not see similar issue, but wanted to just point out. 

```
 public static String replaceSpecialCharacterInPropertyValue(String value) {
    value = StringEscapeUtils.**escapeJava**(value);
```

Reply:
Please add some comments stating this is for backward compatibility, where the old commons library escapes character in a different way